### PR TITLE
Add language dropdown to compose view

### PR DESCRIPTION
**Pull request by Tak**
#2459 

- Add dropdown to compose view
  - iso language code on spinner button
  - language name in current language (language name in target language) on dropdown entries
- Language is included when posting statuses
- Language is saved/restored with drafts
- Compose language defaults to the language chosen for Tusky or the system language

Known issues:
- ~~We're tight on UI space in the compose view bottom bar, maybe need to rearrange~~
- ~~At least one of our translations (norsk bokmål) is keyed with a language code that doesn't match the one android gives it, making the defaulting not work (see [matrix message](https://matrix.to/#/!PWkUGGpeMZWDPmdfRE:matrix.org/$1659728795246423pnsUm:matrix.org?via=m.charlag.space&via=matrix.org&via=tchncs.de))~~
- Dropdown doesn't prioritize recently used languages (always ordered by iso language code)
- General UI guidance that I seem to always need :wink:

### CODE REVIEW

**Comment by connyduck** [comment]
Works quite well already, thx. My main problem with it is that we don't have enough space for the button in the bottom bar.
Either we remove the #/@ and put it there, or we move the avatar into the TextField like Twitter and the Mastodon app do.

**Comment by Tak** [comment]
Yeah, same. We could put it *next to* #@ for now?

**Comment by connyduck** [comment]
> Yeah, same. We could put it next to #@ for now?

yes! please do

**Comment by connyduck** [comment]
(sry I did not want to edit your comment)

**Comment by Tak** [comment]
😄

File: app/src/main/res/layout/activity_compose.xml [hunk 1]
```diff
@@ -22,6 +22,19 @@
             tools:ignore="ContentDescription" />
         <!--content description will be set in code -->
 
+        <Spinner
+            android:id="@+id/composePostLanguageButton"
+            style="@style/TuskyImageButton"
+            android:layout_width="44dp"
```
Comment (connyduck) [inline app/src/main/res/layout/activity_compose.xml:28]: I think it needs to be set to something larger or even `wrap_content`

File: app/src/main/java/com/keylesspalace/tusky/adapter/LocaleAdapter.kt [hunk 2]
```diff
@@ -0,0 +1,28 @@
+package com.keylesspalace.tusky.adapter
```
Comment (connyduck) [inline app/src/main/java/com/keylesspalace/tusky/adapter/LocaleAdapter.kt:1]: add the Tusky license header please

**Review by connyduck — COMMENTED** [review]
Works very well 😍

File: app/src/main/res/layout/activity_compose.xml (hunk 1 above)
Comment (Tak) [reply app/src/main/res/layout/activity_compose.xml:28]: Argh, what language is that?
The issue with `wrap_content` is that it wants to size itself for the dropdown items instead 😐.
Although if you know of a workaround (or if I'm just Doing It Wrong)…

Comment (Tak) [reply app/src/main/res/layout/activity_compose.xml]: Ok, checked this variation against all languages installed in my emulator

**Review by connyduck — APPROVED** [review]

**Pull request merged**